data-extraction/data-exporter/task/index.js

```diff
@@ -94,52 +94,61 @@ output('quests', () => db('Quest').simpleObject('Name'))
 output('races', () => db('Race').simpleObject('Masculine'))
 
 const craftTypeJobMap = [8, 9, 10, 11, 12, 13, 14, 15]
-output('recipes', () => db('Recipe', false).map(recipe => {
-  if (!+recipe['#'] || !+recipe['Item{Result}']) return null
-  const level = db('RecipeLevelTable', false, true).findById(recipe.RecipeLevelTable)
+output('recipes', () => {
+  const canBeHq = db('Item').toObject(row => row.CanBeHq === 'True')
+  return db('Recipe', false).map(recipe => {
+    if (!+recipe['#'] || !+recipe['Item{Result}']) return null
+    const level = db('RecipeLevelTable', false, true).findById(recipe.RecipeLevelTable)
 
-  const ingredients = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9].map(index => {
-    const id = +recipe[`Item{Ingredient}[${index}]`]
-    const amount = +recipe[`Amount{Ingredient}[${index}]`]
+    const ingredients = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9].map(index => {
+      const id = +recipe[`Item{Ingredient}[${index}]`]
+      const amount = +recipe[`Amount{Ingredient}[${index}]`]
 
-    if (!id || !amount) {
-      return null
-    }
+      if (!id || !amount) {
+        return null
+      }
+
+      return {
+        id,
+        amount,
+        ilvl: id < 20 ? 0 : +db('Item', null, true).findById(id)['Level{Item}']
+      }
+    }).filter(a => a)
+    const totalIlvl = ingredients.reduce((sum, { amount, ilvl }) => sum + amount * ilvl, 0)
+    const maxQuality = Math.floor((+level.Quality) * (+recipe.QualityFactor) / 100)
+    const totalContrib = maxQuality * (+recipe.MaterialQualityFactor) / 100
 
     return {
-      id,
-      amount,
-      ilvl: id < 20 ? 0 : +db('Item', null, true).findById(id)['Level{Item}']
+      id: +recipe['#'],
+      job: craftTypeJobMap[+recipe.CraftType],
+      lvl: +level.ClassJobLevel,
+      yields: +recipe['Amount{Result}'],
+      result: +recipe['Item{Result}'],
+      stars: +level.Stars,
+      qs: recipe.CanQuickSynth === 'True',
+      hq: recipe.CanHq === 'True',
+      durability: Math.floor((+level.Durability) * (+recipe.DurabilityFactor) / 100),
+      quality: maxQuality,
+      progress: Math.floor((+level.Difficulty) * (+recipe.DifficultyFactor) / 100),
+      suggestedControl: +level.SuggestedControl,
+      suggestedCraftsmanship: +level.SuggestedCraftsmanship,
+      controlReq: +recipe.RequiredControl,
+      craftsmanshipReq: +recipe.RequiredCraftsmanship,
+      rlvl: +recipe.RecipeLevelTable,
+      ingredients: ingredients.map(({ id, amount, ilvl }) => ({
+        id,
+        amount,
+        quality: canBeHq[id] ? (ilvl / totalIlvl * totalContrib) : null,
+      })),
+      progressDivider: +level.ProgressDivider,
+      qualityDivider: +level.QualityDivider,
+      progressModifier: +level.ProgressModifier,
+      qualityModifier: +level.QualityModifier,
+      expert: recipe.IsExpert === '1',
```

Comment:
iirc IsExpert is a number not a string, which makes this condition always false, not a big deal but worth a fix later on.